src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,80 @@
+
+package frc.robot.subsystems;
+
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.smartdashboard.*;
+
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.controls.VelocityVoltage;
+
+public class Shooter extends SubsystemBase {
+
+  private final TalonFX pivotMotor = new TalonFX(0); // TODO: Set to pivot motor device Id
+  private final TalonFX leftShootMotor = new TalonFX(0); // TODO: Set to left motor device Id
+  private final TalonFX rightShootMotor = new TalonFX(0); // TODO: Set to right motor device Id
+  private double speed = 0;
+  private VelocityVoltage velocity = new VelocityVoltage(speed);
+  public static boolean motorsAtShootingSpeed = false;
+  private final String smartDashboardSpeed = "Shooter Motor Speed";
+  private final String smartDashboardPosition = "Shooter Motor Speed";
+
+  public enum position {
+    DEFAULT,
+    AMP,
+    SPEAKER;
+  }
+
+  public static position positionOfArm = position.DEFAULT;
+
+  public Shooter() {}
+
+  public void shooterInit() {
+    SmartDashboard.putNumber(smartDashboardSpeed, speed);
+    SmartDashboard.putString(smartDashboardPosition, positionOfArm.toString());
+    leftShootMotor.setInverted(true); // TODO: Change to Right/Left to invert shooting motor
+  }
+
+  public void startMotors(double speed, position position) {
+    this.speed = speed;
+    positionOfArm = position;
+    SmartDashboard.putNumber(smartDashboardSpeed, speed);
+    SmartDashboard.putString(smartDashboardPosition, positionOfArm.toString());
+    if (position == position.AMP) {
+      pivotMotor.setPosition(-0.05);  // TODO: Change to fit amp angle
+    } else if (position == position.SPEAKER) {
+      pivotMotor.setPosition(-0.1);  // TODO: Change to fit speaker angle
+    }
+    rightShootMotor.setControl(velocity.withVelocity(speed));
+    leftShootMotor.setControl(velocity.withVelocity(speed));
+  }
+
+  public void shoot() {
+    if (motorsAtShootingSpeed == true) { // If shooter is in position and ready to fire
+      motorsAtShootingSpeed = false;
+    }
+  }
+
+  public void setDefault() {
+    SmartDashboard.putNumber(smartDashboardSpeed, 0);
+    SmartDashboard.putString(smartDashboardPosition, positionOfArm.toString());
+    rightShootMotor.setControl(velocity.withVelocity(0));
+    leftShootMotor.setControl(velocity.withVelocity(0));
+    if (positionOfArm == position.AMP) {
+      pivotMotor.setPosition(-0.1);  // TODO: Change to fit amp angle
+    } else if (positionOfArm == position.SPEAKER) {
+      pivotMotor.setPosition(-0.05);  // TODO: Change to fit speaker angle
+    }
+    positionOfArm = position.DEFAULT;
+  }
+
+  @Override
+  public void periodic() {
+    if (leftShootMotor.getVelocity().refresh().equals(Math.round(speed/25)*25) && positionOfArm != position.DEFAULT) { // This will effectively give a range of 25 RPM for comparing
```

Comment:
getVelocity returns a `StatusSignal<Double>`. If you look at [the Javadoc](https://api.ctr-electronics.com/phoenix6/release/java/com/ctre/phoenix6/StatusSignal.html), the `refresh` method is the wrong one to call. It returns a `StatusSignal<Double>` that's the refreshed value. You would only need to do this if you were storing the `StatusSignal<Double>` instance itself somewhere.

In our case, the value is refreshed for you and you should just call `getValue()` here. It'll be cached and slightly off timing wise but that doesn't matter at all for our purposes here.

One thing I will point out is that because `refresh()` returns a `StatusSignal<Double>`, and the parameter passed into your `equals` is a double. What that means is that this statement will always return false.

Last thing, I don't believe this code does what you think it does in regards to the "This will effectively give a range of 25 RPM for comparing" comment. Your velocity measurement will be a noisy signal. Your speed will be a more precise number. What this code does is makes sure your speed is a multiple of 25 when comparing and if your velocity isn't exactly that it won't work. Check this code out instead:

```java
double motorVelocity = leftShootMoter.getVelocity().getValue();
motorsAtShootingSpeed = motorVelocity <= speed + 10 && motorVelocity >= speed - 10;
```

This gives us two benefits:
1. It doesn't matter where our arm is, we only care about shooting speed for this variable. But that's actually what we want here. The motors being at shooting speed have nothing to do with the position of the arm. This is why I recommended a `canShoot` method and that method would handle both checking the `motorsAtShootingSpeed` variable and checking the `positionOfArm` variable.
2. `motorsAtShootingSpeed` is now managed automatically. For example, if some other code later is written where the motors spin down but don't actually shoot, our periodic code doesn't need to change at all.